### PR TITLE
Finalise Group Details page

### PR DESCRIPTION
**Pull request by aayush-05**
**Screenshots**
1.
image
2.
image
3.
image
4.
image
5.
image
6.
image
7.
image

**Non-Admin account:**
1.
image
2.
image
3.
image
4.
image

**Note: ** Work done on top of #1485

### CODE REVIEW

**Comment by aayush-05** [comment]
@nitin10s please have a look at this

**Comment by aayush-05** [comment]
Increasing the padding of buttong is shifting the grade section to a little too much towards the right. Shall I continue with that ?

**Comment by nitin10s** [comment]
Can you update the screenshot for correction point 5. I am not able to add multiple members to see that.

**Comment by nitin10s** [comment]
Also, in the Create Assignments Page, the Back button is not styled, you may do that in this PR itself.

**Comment by aayush-05** [comment]
Actually I was refactoring that page on a different branch, so can't do it in this PR. Does it require some styling changes as well? (I found the checkboxes to be changed, minor styling in input boxes border)

**Comment by nitin10s** [comment]
Okay, just update the screenshot for members' cards, to keep the horizontal and vertical margins equal.

**Comment by aayush-05** [comment]
Here you go:
image

**Comment by nitin10s** [comment]
I meant to keep the margins between two horizontal cards and two vertical cards equal. Sorry for the misunderstanding.

**Comment by aayush-05** [comment]
Oh, I misunderstood. Here are the updated screenshots:
image
image

File: .rubocop.yml [hunk 1]
```diff
@@ -247,4 +247,4 @@ Style/ClassAndModuleChildren:
   Enabled: false
 
 RSpec/MultipleExpectations:
-  Max: 3
+  Max: 7
```
Comment (tachyons) [inline .rubocop.yml:250]: This is too big and defeat the purpose of the scope

Comment (aayush-05) [reply .rubocop.yml:250]: Would changing it to 5 do (that would be enough to not violate current tests) or should I revert it back to 3?

Comment (aayush-05) [reply .rubocop.yml:250]: @tachyons

File: app/views/groups/show.html.erb [hunk 2]
```diff
@@ -100,70 +69,285 @@
...
+      <% end %>
+    </tbody>
+  </table>
+  <div class="row">
+    <div class="col-12 groupdetails-back-button-container">
+      <%= link_to 'Back', user_groups_path(current_user) ,class:"anchor-text" %>
+    </div>
+  </div>
+
+  <div id="myModal" class="modal fade" role="dialog">
```
Comment (tachyons) [inline app/views/groups/show.html.erb:313]: Move to rails partial

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -1,96 +1,65 @@
...
+  </div>
+  <div class="row row-text groupdetails-row">
+    <h3>MEMBERS</h3>
+    <% if policy(@group).admin_access? %>
+      <button type="button" class="btn button-primary groupdetails-button-primary-addmember" data-toggle="modal" data-target="#myModal">+ Add Members</button>
+    <% end %>
+  </div>
+  <hr>
+  <div class="row row-text groupdetails-row-scroll">
+    <% @group.users.each do |user| %>
```
Comment (tachyons) [inline app/views/groups/show.html.erb:32]: Move to partial

File: app/views/groups/show.html.erb [hunk 4]
```diff
@@ -100,70 +69,285 @@
...
-                <%= link_to 'Show', group_assignment_path(@group,assignment), class:"assignment-view" %>
-                <% if assignment.status!="closed" %>
-                  <%= link_to 'Edit', edit_group_assignment_path(@group,assignment), class:"assignment-edit-reopen" %>
-                <% else %>
-                  <%= link_to 'Reopen', reopen_group_assignment_path(@group,assignment), class:"assignment-edit-reopen" %>
-                <% end %>
-                <%= link_to 'Destroy', group_assignment_path(@group,assignment),class:"assignment-delete", method: :delete, data: { confirm: 'Are you sure?' } %>
+            </div>
+            <% if policy(@group).admin_access? %>
+              <% if assignment.status!="closed" %>
```
Comment (tachyons) [inline app/views/groups/show.html.erb:102]: `assignment.closed?`

File: app/views/groups/show.html.erb [hunk 5]
```diff
@@ -100,70 +69,285 @@
...
+                  <%= assignment.name %>
+                </div>
+              </td>
+            </tr>
+            <tr class="table-collapsed-assignment-table-empty-row"></tr>
+            <tr>
+              <th class="table-collapsed-head">Deadline</th>
+              <td>
+                <div class="assignment-deadline-date"><%= assignment.deadline %></div>
+                <% if(assignment.deadline > Time.now) %>
```
Comment (tachyons) [inline app/views/groups/show.html.erb:197]: Use Time.current,

File: app/views/groups/show.html.erb [hunk 6]
```diff
@@ -100,70 +69,285 @@
...
+                  <p>(<strong>Time remaining: </strong><%= (( assignment.deadline.to_i - Time.now.to_i)/1.day) %> days,  <%= (( assignment.deadline.to_i- Time.now.to_i)/1.hour)%24 %> hours, <%= (( assignment.deadline.to_i- Time.now.to_i)/1.minute)%60 %> minutes)</p>
```
Comment (tachyons) [inline app/views/groups/show.html.erb:198]: Move this to decorator

**Comment by aayush-05** [comment]
@tachyons made the required changes. Please have a look again

**Pull request merged**